cuppa/src/main/java/org/forgerock/cuppa/Configuration.java

```diff
@@ -31,8 +32,10 @@
     List<Function<TestBlock, TestBlock>> testTransforms = new ArrayList<>();
     TestInstantiator testInstantiator = Class::newInstance;
     Reporter additionalReporter;
+    final Options runState;
 
-    Configuration() {
+    Configuration(Options runState) {
```

Comment:
Im thinking it would be nice for the test transform to know what state it expects and for the `ConfigurationProvider` just to state that the test transform is to be used in a specified order rather than expect the `ConfigurationProvider` to know what state the test transform needs...

Reply:
Yes, this decouples things nicely. Good thinking!